### ISSUE TITLE
Is validate(x::Dict, shcema) is necessary?

### ISSUE THREAD

**Issue by yongheekim-dev** [fredo-dedup/JSONSchema.jl#19]
validate methods seems generic enough to extended for AbstractDict. Is thres reason for allow only Dict for validate?

This is MWE
``` julia 
using JSONSchema
using OrderedCollections

schema = Schema("""{
    "\$schema": "http://json-schema.org/draft-07/schema",
    "\$id": "test.json",
    "type": "object",
    "title": "test",
    "required": [
        "Key"
    ],
    "additionalProperties": true,
    "properties": {
        "Key": {
            "type": "string",
            "enum": ["A", "B"] 
        }
    }
}""")

data = Dict("Key" => "A")
validate(data, schema) 

# validation fail
data2 = OrderedDict("Key" => "A")
validate(data2, schema) 
```

**Comment by odow** [comment]
No reason. Shouldn't be much trouble to go through and relax the type signatures.

**Comment by odow** [comment]
Closed by #20
